src/commands/up.rs

```diff
@@ -40,96 +41,42 @@ pub struct UpCommand {
         )]
     pub bindle: Option<String>,
+
     /// URL of bindle server.
     #[clap(
             name = BINDLE_SERVER_URL_OPT,
             long = "bindle-server",
             env = BINDLE_URL_ENV,
         )]
     pub server: Option<String>,
-}
 
-#[derive(Args, Debug)]
-pub struct UpOpts {
-    /// IP address and port to listen on
-    #[clap(name = ADDRESS_OPT, long = "listen", default_value = "127.0.0.1:3000")]
-    pub address: String,
     /// Temporary directory for the static assets of the components.
     #[clap(long = "temp")]
     pub tmp: Option<PathBuf>,
-    /// Pass an environment variable (key=value) to all components of the application.
-    #[clap(long = "env", short = 'e', parse(try_from_str = crate::parse_env_var))]
-    pub env: Vec<(String, String)>,
-
-    /// The path to the certificate to use for https, if this is not set, normal http will be used. The cert should be in PEM format
-    #[clap(
-            name = TLS_CERT_FILE_OPT,
-            long = "tls-cert",
-            env = TLS_CERT_ENV_VAR,
-            requires = TLS_KEY_FILE_OPT,
-        )]
-    pub tls_cert: Option<PathBuf>,
-
-    /// The path to the certificate key to use for https, if this is not set, normal http will be used. The key should be in PKCS#8 format
-    #[clap(
-            name = TLS_KEY_FILE_OPT,
-            long = "tls-key",
-            env = TLS_KEY_ENV_VAR,
-            requires = TLS_CERT_FILE_OPT,
-        )]
-    pub tls_key: Option<PathBuf>,
-    /// Log directory for the stdout and stderr of components.
-    #[clap(
-            name = APP_LOG_DIR,
-            short = 'L',
-            long = "log-dir",
-            )]
-    pub log: Option<PathBuf>,
-
-    /// Disable Wasmtime cache.
-    #[clap(
-        name = DISABLE_WASMTIME_CACHE,
-        long = "disable-cache",
-        env = DISABLE_WASMTIME_CACHE,
-        conflicts_with = WASMTIME_CACHE_FILE,
-        takes_value = false,
-    )]
-    pub disable_cache: bool,
-
-    /// Wasmtime cache configuration file.
-    #[clap(
-        name = WASMTIME_CACHE_FILE,
-        long = "cache",
-        env = WASMTIME_CACHE_FILE,
-        conflicts_with = DISABLE_WASMTIME_CACHE,
-    )]
-    pub cache: Option<PathBuf>,
-
-    /// Print output for given component(s) to stdout/stderr
-    #[clap(
-        name = FOLLOW_LOG_OPT,
-        long = "follow",
-        multiple_occurrences = true,
-        )]
-    pub follow_components: Vec<String>,
 
-    /// Print all component output to stdout/stderr
-    #[clap(
-        long = "follow-all",
-        conflicts_with = FOLLOW_LOG_OPT,
-        )]
-    pub follow_all_components: bool,
+    /// All other args, to be passed through to the trigger
+    pub trigger_args: Vec<OsString>,
 }
 
 impl UpCommand {
     pub async fn run(self) -> Result<()> {
-        let working_dir_holder = match &self.opts.tmp {
+        let help = self.help;
+        let res = self.run_inner().await;
+        if help && res.is_err() {
```

Comment:
Per your comment: `The help/usage stuff is a little weird but I'm not really sure what the ideal behavior should be. Currently in the PR if you spin up --help, it will attempt to load a manifest and call e.g. spin trigger http --help, but if that fails you'll just get the non-trigger-specific spin up --help usage`

This is tricky. Would it make sense to print the failed command before the general `help`?

Reply:
_I'm not sure._ On one hand, its confusing that `--help` shows different things depending on `spin.toml`. On the other, you don't normally expect to ever see errors with `--help`.

Reply:
Updated to log that error as a warning.